src/datasources/alarms/AlarmsDataSourceCore.ts

```diff
@@ -65,19 +66,30 @@ export abstract class AlarmsDataSourceCore extends DataSourceBase<AlarmsQuery> {
     Object.values(AlarmsQueryBuilderFields).map(field => {
       const dataField = field.dataField as string;
 
-      return [
-        dataField,
-        this.isTimeField(dataField)
-          ? timeFieldsQuery(dataField)
-          : multipleValuesQuery(dataField),
-      ];
+      if (dataField === AlarmsQueryBuilderFields.SOURCE.dataField) {
+        return [dataField, this.getSourceTransformation()];
+      } else if (this.isTimeField(dataField)) {
+        return [dataField, timeFieldsQuery(dataField)];
+      } else {
+        return [dataField, multipleValuesQuery(dataField)];
+      }
     })
   );
```

Comment:
Use switch case with callback variable. 
```suggestion
    Object.values(AlarmsQueryBuilderFields).map(field => {
      const dataField = field.dataField as string;
      let callback;

      switch(dataField) {
        case AlarmsQueryBuilderFields.SOURCE.dataField:
            callback = this.getSourceTransformation();
            break;
      }
      return [dataField, callback];
    })
```